lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
-                  specification[indexName] = {};
-                }
+                  specification[indexName][collectionName] = curatedSpec;
 
-                specification[indexName][collectionName] = curatedSpec;
+                  this.kuzzle.pluginsManager.trigger('log:info', `Validation specification for ${indexName} / ${collectionName} has been loaded.`);
 
-                this.kuzzle.pluginsManager.trigger('log:info', `Validation specification for ${indexName} / ${collectionName} has been loaded.`);
+                  resolve({});
 
-                resolve({});
+                  return null;
+                })
+                .catch(error => {
+                  this.kuzzle.pluginsManager.trigger('log:error', `Specification for the collection ${collectionName} triggered an error`);
+                  this.kuzzle.pluginsManager.trigger('log:error', `Error: ${error.message}`);
 
-                return null;
-              })
-              .catch(error => {
-                this.kuzzle.pluginsManager.trigger('log:error', `Specification for the collection ${collectionName} triggered an error`);
-                this.kuzzle.pluginsManager.trigger('log:error', `Error: ${error.message}`);
+                  resolve({});
 
-                resolve({});
+                  return null;
+                });
+            });
 
-                return null;
-              });
+            promises.push(promise);
           });
-
-          promises.push(promise);
         });
+
+        return Bluebird
+          .all(promises)
+          .then(() => {
+            this.specification = specification;
+            return this.kuzzle.pluginsManager.trigger('log:info', 'All validation specifications have been treated.');
+          });
       });
+  }
 
-      return Promise
-        .all(promises)
-        .then(() => {
-          this.specification = specification;
-          return this.kuzzle.pluginsManager.trigger('log:info', 'All validation specifications have been treated.');
+  /**
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {CollectionSpecification} collectionSpec
+   * @param {boolean} verboseErrors
+   * @returns {Promise<boolean>}
+   */
+  isValidSpecification(indexName, collectionName, collectionSpec, verboseErrors) {
+    // We make a deep clone to avoid side effects
+    const specification = _.cloneDeep(collectionSpec);
+    verboseErrors = verboseErrors || false;
+
+    return new Bluebird(resolve => {
+      this
+        .curateCollectionSpecification(indexName, collectionName, specification, true, verboseErrors)
+        .then(result => {
+          if (verboseErrors) {
+            if (result.isValid === false) {
+              return resolve(result);
+            }
+          }
+          return resolve({isValid: true});
+        })
+        .catch(error => {
+          // we do not want to reject since this method goal is to know what is going wrong with the given spec
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 111 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)